test/createAzureResource.test.ts

```diff
@@ -0,0 +1,104 @@
+        const testInputs: string[] = [appName, 'Linux', 'Node.js 10.10 (LTS - Recommended for new apps)'];
+        ext.ui = new TestUserInput(testInputs);
+
+        await vscode.commands.executeCommand('appService.CreateWebApp');
+        const client: WebSiteManagementClient = getWebsiteManagementClient(testAccount);
+        const defaultRgName: string = 'appsvc_rg_linux_centralus';
+        const createdApp: WebSiteManagementModels.Site = await client.webApps.get(defaultRgName, appName);
+        assert.ok(createdApp);
+
+        ext.ui = new TestUserInput([appName, DialogResponses.deleteResponse.title, DialogResponses.yes.title]);
+        await vscode.commands.executeCommand('appService.Delete');
+        const deletedApp: WebSiteManagementModels.Site | undefined = await client.webApps.get(defaultRgName, appName);
+        assert.ifError(deletedApp); // if app was deleted, get() returns null.  assert.ifError throws if the value passed is not null/undefined
+    });
+
+    test('Create and Delete New Web App (Advanced)', async () => {
+        const resourceName: string = getRandomHexString().toLowerCase();
+        resourceGroupsToDelete.push(resourceName);
+        await vscode.workspace.getConfiguration(constants.extensionPrefix).update('advancedCreation', true, vscode.ConfigurationTarget.Global);
+
+        const testInputs: string[] = [resourceName, '$(plus) Create new resource group', resourceName, 'Linux', 'Node.js 10.10 (LTS - Recommended for new apps)', '$(plus) Create new App Service plan', resourceName, 'B1', 'West US'];
+        ext.ui = new TestUserInput(testInputs);
+
+        await vscode.commands.executeCommand('appService.CreateWebApp');
+        const client: WebSiteManagementClient = getWebsiteManagementClient(testAccount);
+        const createdApp: WebSiteManagementModels.Site = await client.webApps.get(resourceName, resourceName);
+        assert.ok(createdApp);
+
+        ext.ui = new TestUserInput([resourceName, DialogResponses.deleteResponse.title, DialogResponses.yes.title]);
+        await vscode.commands.executeCommand('appService.Delete');
+        const deletedApp: WebSiteManagementModels.Site | undefined = await client.webApps.get(resourceName, resourceName);
+        assert.ifError(deletedApp); // if app was deleted, get() returns null.  assert.ifError throws if the value passed is not null/undefined
+    });
+});
+
+function getWebsiteManagementClient(testAccount: TestAzureAccount): WebSiteManagementClient {
+    return new WebSiteManagementClient(testAccount.getSubscriptionCredentials(), testAccount.getSubscriptionId());
+}
+
+function getResourceManagementClient(testAccount: TestAzureAccount): ResourceManagementClient {
+    return new ResourceManagementClient(testAccount.getSubscriptionCredentials(), testAccount.getSubscriptionId());
+}
+
+function getRandomHexString(length: number = 10): string {
```

Comment:
You've also got this function here: https://github.com/Microsoft/vscode-azureappservice/blob/master/src/commands/deploy.ts#L188

Time to put it in a shared location